rewrite-maven/src/test/java/org/openrewrite/maven/search/DependencyInsightDependencyGraphTest.java

```diff
@@ -0,0 +1,301 @@
+            .dataTable(DependenciesInUse.Row.class, rows -> {
+                assertThat(rows).hasSize(1);
+                DependenciesInUse.Row row = rows.getFirst();
+                assertThat(row.getDependencyGraph()).isEqualTo(
+                  """
+                    jakarta.annotation:jakarta.annotation-api:1.3.5
+                    \\--- org.glassfish.jersey.core:jersey-server:2.35
+                         \\--- compile
+                    """.strip());
+                assertThat(row.getDepth()).isEqualTo(1);
+            })
+            .recipe(new DependencyInsight("jakarta.annotation", "jakarta.annotation-api", null, null, null)),
+          pomXml(
+            """
+              <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <dependencies>
+                  <dependency>
+                      <groupId>org.glassfish.jersey.core</groupId>
+                      <artifactId>jersey-server</artifactId>
+                      <version>2.35</version>
+                  </dependency>
+                </dependencies>
+              </project>
+              """,
+            """
+              <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <dependencies>
+                  <!--~~(jakarta.annotation:jakarta.annotation-api:1.3.5)~~>--><dependency>
+                      <groupId>org.glassfish.jersey.core</groupId>
+                      <artifactId>jersey-server</artifactId>
+                      <version>2.35</version>
+                  </dependency>
+                </dependencies>
+              </project>
+              """
+          )
+        );
+    }
+
+    @Test
+    void dependencyGraphWithNullScopeSearchesAllScopes() {
+        rewriteRun(
+          spec -> spec
+            .dataTable(DependenciesInUse.Row.class, rows -> {
+                assertThat(rows).hasSize(2);
+                DependenciesInUse.Row row1 = rows.get(0);
+                // When scope is null, all scopes are searched
+                // Dependencies without an explicit scope default to compile
+                assertThat(row1.getDependencyGraph()).isEqualTo(
+                  """
+                    com.google.guava:guava:29.0-jre
+                    \\--- compile
+                    """.strip());
+                assertThat(row1.getDepth()).isEqualTo(0);
+                DependenciesInUse.Row row2 = rows.get(1);
+                assertThat(row2.getDependencyGraph()).isEqualTo(
+                  """
+                    com.google.guava:guava:29.0-jre
+                    \\--- compile
+                    """.strip());
+                assertThat(row2.getDepth()).isEqualTo(0);
+            })
+            .recipe(new DependencyInsight("com.google.guava", "guava", null, null, null)),
```

Comment:
We're now seeing this test occasionally fail on main, as it alternates between finding `compile` and `test` scope. Would you mind running this a few times and figuring out what's needed to make it stable?